scanpy/external/tl/_harmony_timeseries.py

```diff
@@ -0,0 +1,462 @@
+"""Harmony time series for data visualization with augmented affinity matrix at discrete timepoints
+"""
+
+from scanpy import logging as logg
+import pandas as pd
+from collections import OrderedDict
+from itertools import chain
+import numpy as np
+import os
+
+
+def harmony_timeseries(data: list, **kargs):
+    """
+    Harmony time series for data visualization with augmented affinity matrix at discrete timepoints [Nowotschin18i]_.
```

Comment:
Please rebase against master – I lifted the line length to black’s default of 88 chars. This and many other lines are still too long.